Sources/FoundationEssentials/JSON/BufferView.swift

```diff
@@ -100,7 +102,6 @@ extension BufferView:
     RandomAccessCollection {
 
     typealias Element = Element
-    typealias Index = BufferViewIndex<Element>
```

Comment:
I agree that the definitions below should be made in terms of `Index`, but I disagree with moving the definition of `Index` from here to the main struct. `Index` is part of the `Collection` conformance, while the struct simply holds two stored properties. While one of these _is_ `count`, if we were to change the internal representation of the struct to be two indices, I'd want to move the computed `count` to be among the `Collection` conformance.

Reply:
@glessard Thank you for your review. The following changes were added.

* revert Index
https://github.com/apple/swift-foundation/pull/96/commits/6900c61344048242878ccbb9b79b0a47b2b75e80

* move the computed count
https://github.com/apple/swift-foundation/pull/96/commits/6900c61344048242878ccbb9b79b0a47b2b75e80